# Meeting 2017-11-02

This meeting was with Avinash only.

We discussed the following subjects:

- Quick overview of PINDEL, BreakDancer and Delly.
- Quck review of the data (1000GP phase 3).
- Pipelines used at NYGC.
- Data used in actual genome centers.

## PINDEL, DELLY and Breakdancer

Overview will be done in a dedicated note.

## Data from the 1000GP phase 3

- High quality
- Freely available for research purposes
- Curated by using multiple callers
- Done by the 1000GP (so trustworthy and likely to become a standard).

| SV Class  | No. variants |
|-----------|--------------|
|deletion   | 42.279       |
|duplication| 6.025        |
|mCNV       | 2.929        |
|Inversion  | 786          |
|MEI        | 16.631       |
|NUMT       | 168          |


## Pipelines used at NYGC

non cancer patient: GenomeSTRIP

cancer patient: ASCAT, DELLY, PINDEL, BreakDancer

## Data used in actual genome center

Most often the data used in SV caller papers is the one from the 1000GP.

They often assume a coverage of 4x up to 10x with 80-150bp reads with paired ends.

The actual coverage is closer to 30X in actual sequencing data (coverage is lower for population
studies).

The range of prices is the following:

- ~40$ for arrays of the exome
- ~100$ for an omni array data (illumina)
- ~1k-2k$ for WGS at 30x with illumina
- ~600$ for BGI sequencing (no idea what it is though)

He also mentionned an interesting paper for sequencing at ultra low coverage (~2-3x) whose reference
I lost (will update as soon as I find it again).

It is interesting to know that de novo mutation usualy are responsible for up to 120bp, thus finding
a large amount of de novo mutations is usualy a very bad sign (for the algorithm).

## Where to look at for papers

Journals:

- Nature Methods
- Genome Research
- Bioinformatics

Conferences:

- ASHG
- AACR

Online resources:

- Twitter (just like in DL, follow the right people)
- Biostars

## Textbooks

- Human molecular genetics
- Introduction to bioinfo (Arthur M. Lesk)
- Eric Lander's MOOC

These are extremely focused on bioinformatics and genetics, but since genomic data is not natural to
humans, it may be worth it to catch up on that area (for example knowing what oxford nanopore is,
chipSeq, rnaSeq ans so on).

## Innovation

According to Avinash, having a tool that is as good as the other state of the art (but with another
method) is already a nice contribution, aiming for a paradigm changing tool is not necessary (even
though it would obviously be better).

Thus the criteria for success will be: Are we as good as the other tools ?
