# Hello stranger

Well the last two weeks were a bit weird so i didn't take a lot of notes.

The few important points are:

- I managed to grab an internship at Google Brain with one of the co-authors of DeepVariant,
  the subject is still to be defined, but it is likely to end up being open source so I'll be able
  to talk about it in a while.
- Yanlei was convinced by the general algorithm for DeepSV, the only step left is to write down
  the details for the generation of the window, i.e. what are the properties of the variants that
  are caught in the window.
- Yanlei seemed to be OK with the strategy is was offering for the publication strategy:
  - First we attempt to present the classifier at an ML conference (probably NIPS)
  - Second we make an actual full fledged SV caller that is usable by bioinformaticians, that one
    I expect to be a technical note in *Bioinformatics* should things go well.


I also got a very good idea, or at least I hope so. We could also train our model on variants
detected with other technologies such as long reads or DNA microarrays, because maybe there
is enough signal in PR to detect them even if a formal algorithm has not been found yet.

# Properties of the data

Property 1: the distance between two reads of a pair follows a normal distribution.
Property 2: the size of the reads follows a normal distribution.

## Types of signals

There are basically 4 signals used for structural variant calling:

- Assembly based, which are rare outside of long read technologies
- Read count, rarely used alone and only works for imbalanced variants. It basically consists
  of looking at the number of reads at a particular location, more reads implies duplication, less
  reads implies deletion.
- Split reads, use unmapped reads and try to map them around suspected variants (by splitting them
  in two or three if needed).
- Paired reads, use pairs of mapped reads. By looking at the distance between them and their
  orientation we can guess when there are likely SVs

In the case of deepSV we plan on using PR and RC signals for scoring the variants, SR will only
be used for breakpoint location.

## Anomalous paired reads

Anomalous read pairs are read pairs that do not have the normal orientation or distance.
These reads need to happen around a breakpoint, and have one read on each side.

Indeed, if one read was on the breakpoint, it could not be mapped (it can technically be only
a little bit on the breakpoint, depending on the mapping quality).

Respectively, if both reads were on a side of a breakpoint, they woulld be mapped normaly and would
thus not be caught as anomalous.

Lemna 1: With anomalous read pairs there is, by consruction, a breakpoint between the two reads.

## Deletion

In the case of deletions there is a single breakpoint in the target genome, that is where the deltion
happens.

ex:

```
target:  ---------       b       -------------
ref:     --------- b1 ******* b2 -------------

where:
b is breakpoint
**** is deleted region.
```

Here reads around the breakpoint will be mapped farther appart that expected on the reference
genome.

Lemna 2: Once again by definition of anomalous read pairs, there is one read on each side of the breakpoint b,
thus all reads in anomalous read pairs will be at most `read_size + insert_size` away from the
breakpoint and as such the deleted region.

Remembering properies 1 and 2, this tells us that we can have a boundary of fixed size in which
we can expect to catch all anomalous read pairs around a breakpoint.

Lemna 3: Respectively, should we have a group of anomalous read pairs supporting the same deletion
we can say with a known confidence interval, that most anomalous read pairs to the left of the breakpoint b1
will be contained within `insert_size + read_size` of the beginning of the leftmost supporting read. The same
is trivially true for the reads on the right of the breakpoint b2.

## Inversion

ex

```
target: ->->->->->->->-> b1 <-<-<-<-<-<-<- b2 ->->->->->->->->
ref:    ->->->->->->->->    ->->->->->->->    ->->->->->->->->

where b1 and b2 are the two breakpoints
```

Here anomalous read pairs are pairs where one read is in the inversed region and the other
is in the regular one (i.e. either one read left of b1 or one read right of b2). Indeed reads
between b1 and b2 will be mapped properly (because BWA is not sensitive to the order of the reads).

Reads in the inversed region will be mapped in the wrong direction, and be mapped farther away
than expected (as the region is inversed they will be moved closer to the other breakpoint).

We can see that if one read is to the left of the of b1, the other one will be mapped close to
b2 inside the inversed region and in the wrong direction, and that if one read is to the right
of b2, the other one will be mapped close to b1 in the invversed region and in the wrong direction
too.

We can also see by using properties 1 and 2, that reads outside de inversed region will be at most
`insert_size + read_size` away from the breakpoint and that the same is true for the reads inside
the inversed region.

Lemna 4: Anomalous read pairs supporting an inversion will be distributed around the two
breakpoints within `2*(insert_size + read_size)`.

Lemna 3: Respectively, should we have a group of anomalous read pairs supporting the same inversion
we can say with a known confidence interval, that most anomalous read pairs around the left breakpoint
will be contained within `2 * (insert_size + read_size)` of the beginning of the leftmost supporting
read. The same is trivially true for the right breakpoint.

## Tandem Duplication

ex:
```
target: -------- b1 +++++++ b +++++++ b1 ----------
ref:    --------    d1   +++++++   d2    ----------
```

Here anomalous read pairs, are pairs that have one read on each side of breakpoint b, indeed
read pairs fully contained between b1 and b or b and b2, will be mapped  properly on the
duplicated region.

lemna 6: These reads will both be aligned in the wrong direction (`<- ->` instead of `-> <-` for example),
and inside the duplicated region, one within `insert_size + read_size` to the right of d1, the
other within `insert_size + read_size` to the left of d2.
The read count in the duplicated region will be about the double of outside the duplicated region.

## Tandem CNVs

Lemna 7: It  trivially follows that should a region be duplicated more than once, the anomalous read pairs
will be distributed the same way as with tandem duplication (except for higher read count and more
anomalous read pairs).


## Conclusion

Theorem 1: Putting all the lemnas together we can conclude that given a set of structural, there are two windows
of size `2 * (insert_size + read_size)` each, centered on the two breakpoints in the reference genome
that contains most of the anomalous read pairs that will support it.

Theorem 2: Given a type of variant and a set of anomalous read pairs, we can construct two windows of size
`2 * (insert_size + read_size)` that will contain the breakpoints in the reference genome.

Note: for tandem CNVs and tandem CNVs, windows of size `insert_size + read_size` would have sufficed to contain
all the anomalous read pairs, however they would not have contained reads outside of the duplicated area,
which would make us loose the RC information.
Taking larger windows also allows us to have a single size for all our windows, and thus using the same model
for all the variants.

Note: All the window sizes given here assumed that the reads and inserts were of the same size, when
we know that they are variable, adding a few standards deviations will allow us to catch the read pairs
with higher insert size or read size.
