# Data sources

The FDA precision challenge uses the Genome in a Bottle data, both HG001 and HG002

The website for this dataset is hosted [here](http://jimb.stanford.edu/giab/).

They seem to have a benchmarking stantard [here](https://github.com/ga4gh/benchmarking-tools/)
which is beeing worked on by Global Aliance for Genomics and Health.

Data seems hosted on this [FTP](ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/release).

Data is described [here](https://www.nature.com/articles/sdata201625.pdf).

# Benchmarks papers

http://www.nature.com/nrg/journal/v18/n8/full/nrg.2017.44.html

# File formats

## FASTA file

Simply a sequence of nucleic acid or peptides in ASCII format preceded by a description
(starting with `>` or `;`)

ex:
```
>SEQUENCE_1
MTEITAAMVKELRESTGAGMMDCKNALSETNGDFDKAVQLLREKGLGKAAKKADRLAAEG
LVSVKVSDDFTIAAMRPSYLSYEDLDMTFVENEYKALVAELEKENEERRRLKDPNKPEHK
IPQFASRKQLSDAILKEAEEKIKEELKAQGKPEKIWDNIIPGKMNSFIADNSQLDSKLTL
MGQFYVMDDKKTVEQVIAEKEKEFGGKIKIVEFICFEVGEGLEKKTEDFAAEVAAQL
```

## FASTQ file

Basically a FASTA file with base quality score associated to it.

- The `>` or `;` are replaced with `@`
- The third line is just a `+`
- Fourth line is the quality score in log, expressed in ASCII (`!` lowest and `~` highest).

ex:
```
@SEQ_ID
GATTTGGGGTTCAAAGCAGTATCGATCAAATAGTAAATCCATTTGTTCAACTCACAGTTT
+
!''*((((***+))%%%++)(%%%%).1***-+*''))**55CCF>>>>>>CCCCCCC65
```

## VCF file

Variant Call Format

# Alignment

GRCh37 and GRCh38 are reference to the build of the human reference genomes.
Results are given in these coordinates (thus why FDA insisted on the reference, to avoid aligning on
the wrong one.)

# Conferences

Following a discussion with lau I checked out how deep learning faired in Bioinformatics, seems to
get decently published.

Currently reading
[two](https://academic.oup.com/bioinformatics/article/33/17/2675/3791808/A-deep-learning-framework-for-improving-long-range)
deep learning
[papers](https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btx264)
published there to see whether the review is decent and DL oriented.

# SNP vs SNV

Polymorphism needs to be present in at least 1% of the population, SNV has no restrictions on the
frequency. See
[here](https://www.quora.com/Genomics-What-is-the-difference-between-an-SNP-and-an-SNV).
