Written on 2018-01-10.

# ICML papers

## W. H. Dempsey et al., “iSurvive: An Interpretable, Event-time Prediction Model for mHealth,” in PMLR, 2017, pp. 970–979.

Work on CT-HMM: Continuous Time HMMs, which are HMMs where transitions from state to state happen on
a continuous way.

They adapt EM for their CT-HMM (did previous work to introduce CT-HMM in 2015 at NIPS), and say that
HMMs are useful for interpretation (but what the staes are is chosen by the algo i believe).

## M. Zhao, S. Yue, D. Katabi, T. S. Jaakkola, and M. T. Bianchi, “Learning Sleep Stages from Radio Signals: A Conditional Adversarial Architecture,” in PMLR, 2017, pp. 4100–4109.

Pretty simple conv+rnn model with a GAN, put some maths to prove a trival result about GANs (or so I
believe because I suck at GANs and found that trivial).

There results on the other hand are interesting because they managed to use a GAN architecture to
remove noise from the representation which is great for the application domain and generalization
power.

It interests us for some points:

- puts results foremost and not to heavy handed on the maths proofs
- put some t-sne to prove the model can separate labels nicely in 2d (good for us)
- Citation is pretty heavy handed regarding ML papers (20% bio).

### Structure

- 2.5 intro + related
- 3 pages methods + model
- 2.5 experiments + metrics + rationale

- The model is really for show in my opinion, but ma have been necessary.
- 1 page of figures
- very little metrics in my opinion (I may be biased for too much metrics on the other hand).

# N. Spies, J. M. Zook, M. Salit, and A. Sidow, “svviz: a read viewer for validating structural variants,” Bioinformatics, vol. 31, no. 24, pp. 3994–3996, Dec. 2015.

*Note*: It is interesting that they only show the reads supporting a specific haplotype, there may
be some very good reasons for that (for ex DUP/DEL hetalt would be seen as homref) need to
investigate further.

They say that putting all reads close to the putative SV creates a lot of noise which is hard for a
human to detect, it may be different for an algo.

It is noteworthy that they are very long at treating a single location ~10-30s which is an order of
magnitude higher that what we should do (then again we will do batch and them do not, the bottleneck
may be IOs).

It is noteworthy that they only work for genotyping and not discovery, indeed the require a VCF of
known SVs, this is how they only show reads supporting an SV. Indeed they try to realign the reads
to possible alleles at the location and assign them to the most likely allele (requires knowing in
advance the alleles).

Need to look at `E. Halper-Stromberg, J. Steranka, K. H. Burns, S. Sabunciyan, and R. A. Irizarry,
“Visualization and probability-based scoring of structural variants within repetitive sequences,”
Bioinformatics, vol. 30, no. 11, pp. 1514–1521, Jun. 2014.` as it may have interesting insights
(this is the previous work they build on).


# DeepSV

Tried to analyse the VCFs from 1000GP and failed (more on 2018-01-10 note).
