# Intro

Yesterday's plan was not put in action because of Ikea.

# Deep learning in bioinformatics

Perusing through the papers published in [Bioinformatics (Oxford Press)](https://academic.oup.com/bioinformatics)
showed that many papers are published with deep learning mathods (mostly in proteins).

Population genomics uses more standard methods (gaussian mixtures, ensemble methods, naive bayes..).

But it is interesting to note that the journal is open to these methods, even though they are not
used a lot in out area.

# Opprotunities in deep learning

I finally managed to read that paper, or at least the parts relevant to the PhD.

Basically, a lot can be done and not a lot of groups work on that: Poplin did deep variant and
another group did their framework.

The [framework](http://www.biorxiv.org/content/biorxiv/early/2016/12/30/097469.full.pdf)
of the other group does not inspire me at all, they build their own features which
seems odd for a deep learning pipeline. They kept on comparing themselves to Poplin and it was
extremey awkard.
I expect this to be extremely unreliable and a not the way to go.
What's more, their choice of implementation seems odd at best (no one serious uses deep4j)

Note: the team even says their previous results are bad in a followup paper while being careful not
to cite Poplin...

The review mentionned the idea of improving the representation, which conforms me in my idea that
this is the way to go.

Precision and other metrics being measured at 90% (and having crazy results there), it seems to me
that going to 95% or even 100% would be a good target. It would furthermore be usefull to catch
rarer genetic variants.

NOTE: explain what these % mean.

# SNP or SNV caller ?

