# deepSV preproc

Wrote 2 unit tests for the parsing, everything is in order.

*Note*: all recadords in the vcf do not follow the same format, this is extremely odd (`CIPOS` and
`CIEND` that sometimes happen)

```
VCFRecord { chromosome: "1", pos: Some(4204668), id: "essv16952676", reference: "C", alt: [DEL], quality: None, filter: ".", info: [DBVARID, CALLID("DEL_pindel_129_NA12878"), SVTYPE("DEL"), EXPERIMENT(9), SAMPLE("NA12878"), END(4204717), REGION("esv3818201")] }
VCFRecord { chromosome: "1", pos: Some(7570074), id: "essv16956757", reference: "A", alt: [DEL], quality: None, filter: ".", info: [DBVARID, CALLID("BI_GS_DEL1_B5_P0008_654_NA12878"), SVTYPE("DEL"), EXPERIMENT(9), SAMPLE("NA12878"), CIPOS(0, 0), CIEND(0, 0), END(7571521), REGION("esv3818279")] }
VCFRecord { chromosome: "1", pos: Some(9595360), id: "essv16960381", reference: "A", alt: [DEL], quality: None, filter: ".", info: [DBVARID, CALLID("YL_CN_GBR_48_NA12878"), SVTYPE("DEL"), EXPERIMENT(9), SAMPLE("NA12878"), CIPOS(-1000, 500), CIEND(-500, 1000), END(9597869), REGION("esv3818337")] }

```

# PicardTools for metrics

In order to get the metrics we want here are the two commands in PicardTools that we need:

- [AlignmentSummaryMetrics](https://broadinstitute.github.io/picard/picard-metric-definitions.html#AlignmentSummaryMetrics)
- [InsertSizeMetrics](https://broadinstitute.github.io/picard/picard-metric-definitions.html#InsertSizeMetrics)

Their command lines follow:


```
java -jar picard.jar CollectAlignmentSummaryMetrics \
       R=reference_sequence.fasta \
       I=input.bam \
       O=output.txt
```

```
java -jar picard.jar CollectInsertSizeMetrics \
      I=input.bam \
      O=insert_size_metrics.txt \
      H=insert_size_histogram.pdf \
      M=0.5
```
