# DeepSV

## Translocations

DeepSV seems to be pretty bad for translocations (and thus MEI), indeed they require to look at two
regions, which does not fit into the the model of a single window used in DeepSV.

A closer analysis of the SV phase 3, supplementary material shows that MEI are only caught with a
single tool (MELT, and apparently VariationHunter), it thus does not seem to be to big of an issue not to catch them.

Focusing only on CNVs (+ inversions) thus looks like a good trade off, as they are the only ones to
fit into our model.

Note that according to the MELT paper, it was done specifically for the phase 3 release of SVs, thus
the problem was poorly solved before.

Gardner, Eugene J., Vincent K. Lam, Daniel N. Harris, Nelson T. Chuang, Emma C. Scott, W. Stephen
Pittard, Ryan E. Mills, and Scott E. Devine. “The Mobile Element Locator Tool (MELT):
Population-Scale Mobile Element Discovery and Biology.” Genome Research 27, no. 11 (November 2017):
1916–29. https://doi.org/10.1101/gr.218032.116.

Hormozdiari, Fereydoun, Iman Hajirasouliha, Andrew McPherson, Evan E. Eichler, and S. Cenk Sahinalp.
“Simultaneous Structural Variation Discovery among Multiple Paired-End Sequenced Genomes.” Genome
Research 21, no. 12 (December 2011): 2203–12. https://doi.org/10.1101/gr.120501.111.

Sudmant, Peter H., Tobias Rausch, Eugene J. Gardner, Robert E. Handsaker, Alexej Abyzov, John
Huddleston, Yan Zhang, et al. “An Integrated Map of Structural Variation in 2,504 Human Genomes.”
Nature 526, no. 7571 (October 1, 2015): 75–81. https://doi.org/10.1038/nature15394.

# SV calling profiling

For the proper profiling of SV calling tools the references are:

- http://www.brendangregg.com/linuxperf.html: good presentation of perf and its effect.
- `perf mem`: command to have mem info when launching perf

NOTE: launched a `perf record` of delly, with a single thread:

```
➜  delly git:(master) ✗ perf record ./src/delly call -g /data/fraimund/ftp.1000genomes.ebi.ac.uk/vol1/ftp
/technical/reference/phase2_reference_assembly_sequence/hs37d5.fa  /data/fraimund/ftp.1000genomes.ebi.ac.
uk/vol1/ftp/phase3/data/NA12878/alignment/NA12878.mapped.ILLUMINA.bwa.CEU.low_coverage.20121211.bam 
```

# HaplotypeCaller indels size

According to
[geraldine](https://gatkforums.broadinstitute.org/gatk/discussion/3932/haplotypecaller-and-detection-of-large-indels),
the rule of thumb for indels with HC is that it can catch up to half the read size.

We can thus consider indels of that size to be caught by HC (we are limited by 2 * (insert + read) size).

# An Introduction to Deep Learning on Biological Sequence Data: Examples and Solutions.

Jurtz, Vanessa Isabell, Alexander Rosenberg Johansen, Morten Nielsen, Jose Juan Almagro Armenteros,
Henrik Nielsen, Casper Kaae Sønderby, Ole Winther, and Søren Kaae Sønderby. “An Introduction to Deep
Learning on Biological Sequence Data: Examples and Solutions.” Bioinformatics (Oxford, England) 33,
no. 22 (November 15, 2017): 3685–90. https://doi.org/10.1093/bioinformatics/btx531.

Decent paper, the introduction goes into a presentation of deep learning method and they then go on
implementing them to 3 problems.

In their opinion their contribution is to build a set of notebooks and methodes accessible to all
bioinformaticians in order to allow them to use deep learning.

They do not give a lot of thoughts to the data representation and basicaly slap a:
`Conv->FC->LSTM->Attention` on all the problems (and it works).
